### DISCUSSION TITLE
Pasting from clipboard inserts extra characters when terminal has been open for a while

### DISCUSSION THREAD

**Discussion by dpo** [wezterm/wezterm#4773] [General]
Hi. Thank you for Wezterm! I wanted to report an issue I’ve been having with the most recent stable version (20230712-072601) installed via Homebrew on macOS.

I often login to a remote server by copy-pasting my password from a password manager (1Password, but I don’t think that is relevant). I noticed that when I leave a Wezterm tab open for “a while” (a few days), pasting my password from the clipboard inserts extra characters before and after the password. If I immediately open a new Wezterm tab and paste my password there, it is correct (there are no extra characters), and I can log in to the distant server.

Specifically, the extra characters are:
- `00~` at the front
- `01~` at the back.

Those look like they may be escape sequences of some sort. I read issues here related to Wezterm settings about pasting options, but I don’t believe that is the issue here because the problem does not occur in a “fresh” Wezterm tab; only in a “stale” one.

Thank you.

**Comment by dpo** [comment]
Confirming that the bug is still present in the latest nightly (20240111-073635-6c36a4dd) installed via Homebrew.

**Comment by cynicaljoy** [comment]
This is going to be rough for somebody that doesn't have your set up to recreate the issue. Might want to take a look at [Troubleshooting](https://wezfurlong.org/wezterm/troubleshooting.html) and see if there is anything in the debug overlay/logs that can help. Be sure there is no sensitive information before sharing them.  

Also, it reads as if this wasn't an issue with older versions. Are you able to identify which version the issue was introduced?

**Comment by wez** [comment]
FWIW, those escape sequences sound like they are part of [bracketed paste](https://en.wikipedia.org/wiki/Bracketed-paste).
Are you sure it is wezterm that is wonky here and not something to do with the application you are using?
Note that interactive programs such as your shell or editor will enable this mode with an escape sequence, but if they are interrupted and don't cancel the mode, it can be left enabled for applications that don't understand it, which might the case on your remote system if the version of the software there is older, or otherwise doesn't support bracketed paste.
Some versions of shells forget to disable bracketed paste when spawning child processes, which can have similar results.

**Reply by dpo** [reply]
I believe you are right, thank you. It seems that when the ssh connection doesn’t exit cleanly, pasting is wonky at least part of the time.

**Reply by wez** [reply]
If you encounter this again, open the command palette (CTRL-SHIFT-P) then select "Reset the terminal emulation state in the current pane". That will reset various behaviors back to the default initial state in the terminal emulator.
